pkgs/applications/editors/rstudio/default.nix

```
@@ -43,15 +43,15 @@ let
   version =
   "${RSTUDIO_VERSION_MAJOR}.${RSTUDIO_VERSION_MINOR}.${RSTUDIO_VERSION_PATCH}${RSTUDIO_VERSION_SUFFIX}";
   RSTUDIO_VERSION_MAJOR  = "2023";
-  RSTUDIO_VERSION_MINOR  = "09";
-  RSTUDIO_VERSION_PATCH  = "0";
-  RSTUDIO_VERSION_SUFFIX = "+463";
+  RSTUDIO_VERSION_MINOR  = "12";
+  RSTUDIO_VERSION_PATCH  = "1";
+  RSTUDIO_VERSION_SUFFIX = "+402";
```

```suggestion
  version = "2023.12.1+402";
  RSTUDIO_VERSION_MAJOR = lib.versions.major version;
  RSTUDIO_VERSION_MINOR = lib.versions.minor version;
  RSTUDIO_VERSION_PATCH = lib.versions.patch version;
  RSTUDIO_VERSION_SUFFIX = toString (lib.tail (lib.splitString "+" version));
```
when we do it like this, the version number is a lot more readable. Also we could move this down where the inherit uses them.

---

This is so neat! I just discovered lib.versions last week. Thanks!